### ISSUE TITLE
Can't add recipes to a shared book

### ISSUE THREAD

**Issue by Programie** [TandoorRecipes/recipes#1210]
### Version
Version: 1.0.4

### Bug description
Currently, I can add up to 14 recipes to a book. Once this limit has been reached, adding another recipe to the book silently does nothing. I'm successfully applying the "Manage book" dialog but the recipes is not being added.

If there should be a hard limit, at least a hint should be displayed to the user that there can't be added any more recipes to the book. But in my opinion, a limit doesn't make too much sense for me.

**Comment by vabene1111** [comment]
There definetely should not be a limit, this must be a bug. I will see if I can reproduce and ask you for some information otherwise.

**Comment by Programie** [comment]
I just digged a bit deeper and found the issue: There is no limit for books. Another user on my instance added 14 recipes to a book (which is shared with my user). Then I wanted to add another recipe which silently failed.

Opening the browser console and looking at the API request being made gave me some more details: It returned an error 404 "not found".

Then, I manually changed the owner (created by) of that book to my user and trying adding the recipe again and it worked!

So, there seems to be some issue with shared books. Only the owner can edit books while the users it is shared to can only view them. In my opinion, all users should be able to edit books if they have access to the book. Or, alternatively, implement some permissions for books: Some users are able to edit book while some other users are only able to view them and also don't see them in the book selection dropdown in the "Manage Books" dialog.
